core/src/types/metadata.rs

```diff
@@ -404,6 +406,27 @@ impl Metadata {
         self
     }
 
+    /// Is_current of this entry.
+    ///
+    /// Is_current is a boolean that can be used to identify
+    /// if the version of this entry is the latest version.
+    pub fn is_current(&self) -> Option<bool> {
+        self.is_current
+    }
+
+    /// Set is_current of this entry.
+    ///
+    /// For HeadObject without version_id, we will set it to Some(true).
+    ///
+    /// For HeadObject with version_id, we will set it to None.
+    ///
+    /// For ListObjects, we will set all keys to Some(true)
+    ///
+    /// For ListObjectVersions, we will decide the value based on IsLatest.
```

Comment:
```suggestion
    /// Set the `is_current` status of this entry.
    ///
    /// By default, this value will be `None`. Please avoid using this API if it's unclear whether the entry is current.
    /// Set it to `true` if it is known to be the latest; otherwise, set it to `false`.
```